### PR TITLE
Adding JUPITER_SPIRIT Vehicle

### PR DESCRIPTION
**Pull request by sanazfth**
removing DoorSystem

### CODE REVIEW

File: src/startup_checks.cpp [hunk 1]
```diff
@@ -83,6 +85,8 @@ bool AS::Joystick::check_vehicle_type(ros::NodeHandle * nodeH)
 
   if (vehicle_type == LEXUS_RX_450H)
     PublishControl::max_rot_rad = MAX_ROT_RAD_VEHICLE2;
+  else if (vehicle_type == JUPITER_SPIRIT)
+    PublishControl::max_rot_rad = JUPITER_SPIRIT;
```
Comment (JWhitleyAStuff) [inline src/startup_checks.cpp]: This line doesn't make sense. The `MAX_ROT_RAD_*` variables are defined in `globals.h` and this should be one too. The value should be the maximum rotation range of the steering wheel in Radians.

File: src/publish_control_board_rev3.cpp [hunk 2]
```diff
@@ -354,15 +356,16 @@ void PublishControlBoardRev3::publish_lights_horn_wipers_message(const sensor_ms
...
+       vehicle_type == JUPITER_SPIRIT) &&
       controller != HRI_SAFE_REMOTE)
   {
     pacmod_msgs::SystemCmdInt headlight_cmd_pub_msg;
     headlight_cmd_pub_msg.enable = local_enable;
     headlight_cmd_pub_msg.ignore_overrides = false;
 
     // Headlights
-    if (msg->axes[axes[DPAD_UD]] == AXES_MAX)
+    if (msg->axes[axes[DPAD_UD]] == AXES_MAX && msg->buttons[btns[LEFT_BUMPER]] != BUTTON_DOWN)
```
Comment (JWhitleyAStuff) [inline src/publish_control_board_rev3.cpp]: This change should be part of the Door System PR.

**Comment by JWhitleyAStuff** [comment]
Also, CI checks are failing due to linter errors. Run the following to test `roslint` while inside the `pacmod_game_control` folder: `catkin bt --no-deps --make-args roslint`

File: include/pacmod_game_control/globals.h [hunk 3]
```diff
@@ -127,4 +128,4 @@ static std::mutex turn_mutex;
 }  // namespace Joystick
 }  // namespace AS
 
-#endif  // PACMOD_GAME_CONTROL_GLOBALS_H
+#endif  // PACMOD_GAME_CONTROL_GLOBALS_H
```
Comment (icolwell-as) [inline include/pacmod_game_control/globals.h]: Do you need to change some settings in your editor? It seems to be removing newline characters at the ends of the files.

File: launch/pacmod_game_control.launch [hunk 4]
```diff
@@ -12,7 +12,8 @@
   <arg name="pacmod_board_rev" value="2" unless="$(arg is_pacmod_3)" />
   <arg name="pacmod_board_rev" value="3" if="$(arg is_pacmod_3)" />
 
-  <arg name="pacmod_can_hardware_id" default="43029" />
+  <!-- <arg name="pacmod_can_hardware_id" default="43029" /> -->
+   <arg name="pacmod_can_hardware_id" default="31231" />
```
Comment (JWhitleyAStuff) [inline launch/pacmod_game_control.launch]: Please change this back to the default.

File: launch/pacmod_game_control.launch [hunk 5]
```diff
@@ -24,8 +25,10 @@
        INTERNATIONAL_PROSTAR_122
        VEHICLE_4
        VEHICLE_5
-       VEHICLE_6 -->
-  <arg name="pacmod_vehicle_type" default="LEXUS_RX_450H" />
+       VEHICLE_6
+       JUPITER_SPIRIT -->
+  <!-- <arg name="pacmod_vehicle_type" default="LEXUS_RX_450H" /> -->
+   <arg name="pacmod_vehicle_type" default="VEHICLE_5" />
```
Comment (JWhitleyAStuff) [inline launch/pacmod_game_control.launch]: Please change this back to the default.

File: src/publish_control_board_rev3.cpp [hunk 6]
```diff
@@ -65,7 +65,8 @@ void PublishControlBoardRev3::publish_steering_message(const sensor_msgs::Joy::C
   }
 
   float range_scale;
-  if (vehicle_type == VEHICLE_4 || vehicle_type == VEHICLE_6 || vehicle_type == LEXUS_RX_450H)
+  if (vehicle_type == VEHICLE_4 || vehicle_type == VEHICLE_6 || vehicle_type == LEXUS_RX_450H
+                                || vehicle_type == JUPITER_SPIRIT)
```
Comment (JWhitleyAStuff) [inline src/publish_control_board_rev3.cpp]: I'm really surprised this didn't cause a lint error. Please lay out like this:

```
if (vehicle_type == VEHICLE_4 ||
    vehicle_type == VEHICLE_6 ||
    vehicle_type == LEXUS_RX_450H ||
    vehicle_type == JUPITER_SPIRIT)
```

**Pull request merged**